restcomm/restcomm.telephony/src/main/java/org/restcomm/connect/telephony/Call.java

```diff
@@ -599,7 +604,18 @@ private void executeStatusCallback(final CallbackState state) {
 
                 if (parameters != null) {
                     requestCallback = new HttpRequestDescriptor(statusCallback, statusCallbackMethod, parameters);
-                    downloader.tell(requestCallback, null);
+                    if (!ask) {
```

Comment:
why do we make this optional? why dont we use the AskPattern always?

Reply:
@jaimecasero 
There are 4 stage: initial, ringing, answered and completed.

Just only completed state, Call actor will be close as soon as possible and don't wait downloader actor to finish the job. so we have to have some thing hold Call actor there until finish task.

Call actor is always alive for inital and ringing and answered state.